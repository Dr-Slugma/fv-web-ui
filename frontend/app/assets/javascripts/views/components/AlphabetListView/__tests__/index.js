// A11Y
// const { axe, toHaveNoViolations } = require('jest-axe')
// expect.extend(toHaveNoViolations)
// import ReactDOMServer from 'react-dom/server'

// Standard/Unit
import React from 'react'
import ReactDOM from 'react-dom'

// Component to test
import { AlphabetListView } from '..'
const props = {
  handleClick: () => {},
  routeParams: {},
  dialect: {},
  splitWindowPath: [],
  computeCharacters: {},
  computePortal: {},
  fetchDialect2: () => {},
  fetchCharacters: () => {},
  letter: 'a',
}
describe('AlphabetListView', () => {
  // test('Mounts', () => {
  //   // Structure: Arrange
  //   const container = document.createElement('div')
  //   ReactDOM.render(<AlphabetListView {...props} />, container)

  //   expect(container.querySelector('h2').textContent).toMatch('Browse Alphabetically')

  //   // Structure: Act
  //   // Structure: Assert
  // })

  test('Snapshot', () => {
    const container = document.createElement('div')
    ReactDOM.render(<AlphabetListView {...props} />, container)

    const element = container.querySelector('.AlphabetListView')
    expect(element).toMatchSnapshot()
  })

  // test('Accessibility', async() => {
  //   const html = ReactDOMServer.renderToString(<AlphabetListView {...props} />)
  //   const results = await axe(html)
  //   expect(results).toHaveNoViolations()
  // })
})
