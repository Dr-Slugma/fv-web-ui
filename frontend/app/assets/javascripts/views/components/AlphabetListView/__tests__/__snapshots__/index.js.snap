// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlphabetListView Snapshot 1`] = `
<div
  class="AlphabetListView"
  data-reactid=".0"
/>
`;
